devicetypes/smartthings/orvibo-motion-sensor.src/Orvibo-Motion-Sensor.groovy

```diff
@@ -0,0 +1,207 @@
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
+		log.info "BATT METRICS - attr: ${descMap?.attrInt}, value: ${descMap?.value}, decValue: ${Integer.parseInt(descMap.value, 16)}, currPercent: ${device.currentState("battery")?.value}, device: ${device.getDataValue("manufacturer")} ${device.getDataValue("model")}"
+		List<Map> descMaps = collectAttributes(descMap)
+	    def battMap = descMaps.find { it.attrInt == 0x0021 }
+	    if (battMap) {
+	        def map = getBatteryPercentageResult(Integer.parseInt(battMap.value, 16))
+	    }
+	}
+	return map;
+}
+
+private motionHandler(String description){
+	//inactive
+	def isActive = zigbee.translateStatusZoneType19(description)
+	def value = isActive ? "active" : "inactive"
+	if (value == "active") {
+		def timeout = 3
+		if (motionStopTime)
+			timeout = motionStopTime
+		log.debug "Stopping motion in ${timeout} seconds"
+		runIn(timeout, stopMotion)
+	}
+}
+
+
+
+private Map parseIasMessage(String description) {
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	translateZoneStatus(zs)
+}
+
+private Map translateZoneStatus(ZoneStatus zs) {
+	// Some sensor models that use this DTH use alarm1 and some use alarm2 to signify motion
+	return (zs.isAlarm1Set() || zs.isAlarm2Set()) ? getMotionResult('active') : getMotionResult('inactive')
+}
+
+private Map getBatteryPercentageResult(rawValue) {
+	log.debug "Battery Percentage rawValue = ${rawValue} -> ${rawValue / 2}%"
+	def result = [:]
+	if (0 <= rawValue && rawValue <= 200) {
+		result.name = 'battery'
+		result.translatable = true
+		result.value = Math.round(rawValue / 2)
+		result.descriptionText = "${device.displayName} battery was ${result.value}%"
+	}
+	return result
+}
+
+private Map getMotionResult(value) {
+	String descriptionText = value == 'active' ? "${device.displayName} detected motion" : "${device.displayName} motion has stopped"
+	return [
+			name           : 'motion',
+			value          : value,
+			descriptionText: descriptionText,
+			translatable   : true
+	]
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.debug "ping "
+	return zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER, zigbee.ATTRIBUTE_IAS_ZONE_STATUS) + zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021)
+}
+
+def refresh() {
+	log.debug "Refreshing Values"
+	def refreshCmds = []
+	refreshCmds += zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021) +
+	zigbee.enrollResponse()
+	return refreshCmds
+}
+
+def configure() {
+	log.debug "configure"
+	// Device-Watch allows 2 check-in misses from device + ping (plus 1 min lag time)
+	// enrolls with default periodic reporting until newer 5 min interval is confirmed
+	sendEvent(name: "checkInterval", value:20 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
The comment above does not apply to this device.

Please add another minute to this value.